python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -534,3 +644,80 @@ class BaseRandomForestModel(Base, InteropMixin):
                 align_bytes=align_bytes,
             )
         return fil_model
+
+    @abstractmethod
+    def _compute_oob_score_metric(self, y_true, oob_predictions, valid_mask):
+        """
+        Compute the OOB score metric for the specific model type.
+
+        Parameters
+        ----------
+        y_true : array-like
+            True labels
+        oob_predictions : array-like
+            OOB predictions (1D for regression, 2D for classification)
+        valid_mask : array-like
+            Boolean mask indicating which samples have OOB predictions
+
+        Returns
+        -------
+        float
+            The computed score
+        """
+        pass
```

Comment:
IMO base class methods should raise if not implemented in the base class

```suggestion
        raise NotImplementedError
```

Reply:
Actually, instead of making this a method on the base class, why not dispatch to `r2_score` or `accuracy_score` in `_compute_oob_score` based on `_estimator_type`? You already branch in there anyway on that attribute, would save some LOC and and additional method definition.